### DISCUSSION TITLE
Script-based schedules

### DISCUSSION THREAD

**Discussion by AngryCarrot789** [pelican-dev/panel#154] [Ideas]
Creating schedules in pterodactyl is tedious as you have to navigate the UI to add a single new action (which, most of the time, will be server commands). Having a basic scripting language for a schedule would be very convenient (not to say the pterodactyl system should be removed, since it's easy to use for beginners)

This would be a script example of restarting a Minecraft server:
```
command("broadcast Restarting in 3 seconds...");
// maybe could add time unit parsing, e.g. 3s or 3000ms?
delay(3);
command("kick * Server is restarting!"); // kick all users with a restart message
power("restart"); // "start", "stop", "restart", "terminate"
```

**Comment by lancepioch** [comment]
Thanks, we will look at simplifying and extending the functionality of schedules in the future, however I don't believe that is the way we will go.
